src/npctalk.cpp

```diff
@@ -518,6 +548,71 @@ void game::chat()
             available[npcselect]->talk_to_u();
             break;
         }
+        case NPC_CHAT_EMOTE_OVERLAY: {
+            uilist emenu;
+            emenu.text = std::string( _( "Emote what status effect?" ) );
+
+            // category: utility
+            emenu.addentry( 'a', true, 'a', _( "Clear" ) );
+
+            std::vector<efftype_id> all_effects = find_all_effect_types();
+            char key = 'b';
+
+            // This map will hold dynamic emote entries
+            std::map<efftype_id, std::string> dynamic_emote_map;
+            std::map<int, efftype_id> dynamic_emote_key_map; // Map from menu key (int) to efftype_id
+
+            // Iterate through all effect types to create dynamic emote entries
+            for( const efftype_id &effect_id : all_effects ) {
+                const effect_type &etype = effect_id.obj();
+                std::string effect_str = effect_id.str();  // This gets the string ID.
+
+                if( effect_str.length() >= 6 && effect_str.rfind( "_emote" ) == effect_str.length() - 6 ) {
+                    // Create a temporary effect to get the display name
+                    effect temp_effect( &etype, 0_turns, bodypart_str_id::NULL_ID(), 1, calendar::turn_zero );
+                    std::string display_name = temp_effect.disp_name();
+
+                    // Remove the "(emote)" suffix if present
+                    const std::string emote_suffix = " (emote)";
+                    if( display_name.size() >= emote_suffix.size() &&
+                        display_name.compare( display_name.size() - emote_suffix.size(), emote_suffix.size(),
+                                              emote_suffix ) == 0 ) {
+                        display_name.erase( display_name.size() - emote_suffix.size() );
+                    }
+
+                    // Dynamically add to the emote map
+                    dynamic_emote_map[effect_id] = display_name;
+                    dynamic_emote_key_map[key] = effect_id; // Map the key to the efftype_id
+
+                    // Add the entry to the menu dynamically
+                    emenu.addentry( key, true, key, _( display_name ) );
+
+                    // Increment the key for the next emote
+                    key++;
+                }
+            }
+
+            emenu.query();
+
+            if( emenu.ret < 0 ) {
+                return;
+            }
+
+            efftype_id selected_effect_id;
+            if( dynamic_emote_key_map.find( emenu.ret ) != dynamic_emote_key_map.end() ) {
```

Comment:
```suggestion
            if( dynamic_emote_key_map.contains( emenu.ret ) ) {
```

use [`map::contains` (c++20)](https://en.cppreference.com/w/cpp/container/map/contains)